### ISSUE TITLE
Support NaN and Infinity text constants

### ISSUE THREAD

**Issue by maurobringolf** [xtuc/js-webassembly-interpreter#46]
The text format supports the following literals for floating point numbers

* `nan`: https://github.com/WebAssembly/spec/blob/master/test/core/float_literals.wast#L5-L7
* `nan:0x...`: https://github.com/WebAssembly/spec/blob/master/test/core/float_literals.wast#L8-L13
* `inf`: https://github.com/WebAssembly/spec/blob/master/test/core/float_literals.wast#L14-L16

Spec is here https://webassembly.github.io/spec/core/text/values.html#text-float

**Comment by xtuc** [comment]
We should also figure out how to represent those?

(you are the float expert here), for example:

`(f32.const nan:0x200000)`

could be represented as:

```js
FloatLiteral = {
  type: 'FloatLiteral',
  value: 0x200000,
  hasNaN: true, // i'm probably missing different types of NaN
}
```

Currently it's a NumberLiteral which can be problematic.

**Comment by maurobringolf** [comment]
Yes, thats what I started doing in the `nan-payloads` branch. Theyre still number literals and I have a couple of things to fix in JSON (`NaN`, `Infinity` and hex literals). I will try to submit the PR today. 

https://github.com/xtuc/js-webassembly-interpreter/blob/nan-payloads/test/compiler/parsing/fixtures/watf/nan/payload/expected.json
